### PR TITLE
Create binaries using Goreleaser

### PR DESCRIPTION
**Pull request by ofalvai**
### Context

Migrate binary release workflow to Goreleaser. Based on previous PRs in other repos, such as https://github.com/bitrise-io/stepman/pull/307

### Changes

- Introduce `.goreleaser.yml`
- Clean up release script from `bitrise.yml`

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package version
 
-// VERSION ...
-const VERSION = "0.10.4"
```
Comment (lpusok) [inline version/version.go:4]: Are you sure we do not need this version? I think there is a case where the plugin is downloaded from git (when binary release is not found)

File: _scripts/set_version.go
Comment (lpusok) [inline _scripts/set_version.go:1]: I suppose this updated bitrise-plugin.yml, but we bump that manually anyway right?

**Review by lpusok — APPROVED** [review]

File: _scripts/set_version.go
Comment (ofalvai) [reply _scripts/set_version.go:1]: I think so, yes. Running this code requires tons of parameters, it's much easier to replace the version number manually.

File: version/version.go (hunk 1 above)
Comment (ofalvai) [reply version/version.go:4]: Nice, I didn't know about this feature. But I traced it back to the install code and it looks like when a plugin is installed from source, it's version is an empty string: https://github.com/bitrise-io/bitrise/blob/master/plugins/install.go#L276

**Pull request merged**